gopipe.go

```diff
@@ -9,7 +9,7 @@ import (
 Pipe is a single component that processes items. Pipes can be composed to form a pipeline
 */
 type Pipe interface {
-	Process(in chan interface{}, out chan interface{})
+	Process(in interface{}) (out interface{}, next bool)
```

Comment:
what's the benefit to taking the `out` as an input over returning it as an output?